content/en/docs/collector/building/_index.md

```diff
@@ -2,10 +2,70 @@
 title: Building custom components
 description: Instructions on how to build your own collector components
 weight: 90
+cSpell:ignore: mdatagen ottl
 ---
 
 The OpenTelemetry Collector can not only be extended by existing components, but
 also by custom components, that you develop and build on your own. Here you will
 find instructions on how to build some of those components. For additional
 details take a look into the documents contained within the
 [opentelemetry-collector-contrib repository](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/README.md).
+
+There are several
+[classes](https://github.com/open-telemetry/opentelemetry-collector/blob/67d37183e6ac9b7180fefc6dc3a55f2a75c12fba/cmd/mdatagen/main.go#L192)
+of OpenTelemetry components, such as
+
+- `receivers`: Scrapers and Listeners which
+  [ingest data](https://github.com/open-telemetry/opentelemetry-collector/tree/main/receiver#readme),
+  typically used to convert data from an external source to OTLP
+- `exporters`: Ways to
+  [export data](https://github.com/open-telemetry/opentelemetry-collector/tree/main/exporter#readme)
+  to non-OTLP formats, vendor-specific backends, or other Observability tools
+- `processors`: Ways to
+  [process data](https://github.com/open-telemetry/opentelemetry-collector/tree/main/processor#readme)
+  in a pipeline
+- `connectors`:
+  [Connects](https://github.com/open-telemetry/opentelemetry-collector/tree/main/connector#readme)
+  the output (exporter) of one pipeline to the input (receiver) of another in a
+  single component
+- `extensions`: Ways to
+  [augment the collector runtime](https://github.com/open-telemetry/opentelemetry-collector/blob/main/extension/README.md),
+  such as providing
+  [health checks](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/extension/healthcheckextension/README.md)
+- `cmd`: Commands for building and/or maintain the collector
+- `pkg`:
+  [Packages](https://github.com/search?q=org%3Aopen-telemetry+%22class%3A+pkg%22&type=code)
+  for adding functionality to the collector, such as support for
+  [golden tests](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/pkg/golden#readme)
+  or
+  [OTTL functions](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/pkg/ottl#readme)
```

Comment:
I am against promoting the usage of the `pkg` folder outside of contrib in our public documentation. I don't think we have figured out our strategy here yet